controllers/controllers/services/instances/upsi/controller.go

```diff
@@ -125,11 +126,18 @@ func (r *Reconciler) ReconcileResource(ctx context.Context, cfServiceInstance *k
 	}
 
 	if err = r.validateCredentials(credentialsSecret); err != nil {
-		return ctrl.Result{}, k8s.NewNotReadyError().WithCause(err).WithReason("SecretInvalid")
+		cfServiceInstance.Status.LastOperation = services.LastOperation{
+			Type:  "create",
+			State: "failed",
+		}
+		return ctrl.Result{}, k8s.NewNotReadyError().WithCause(err).WithReason("SecretInvalid").WithNoRequeue()
 	}
 
 	log.V(1).Info("credentials secret", "name", credentialsSecret.Name, "version", credentialsSecret.ResourceVersion)
 	cfServiceInstance.Status.Credentials = corev1.LocalObjectReference{Name: credentialsSecret.Name}
+
+	r.setLastOperation(cfServiceInstance, credentialsSecret)
```

Comment:
I would recommend this function to return the `lastOperation` value rather than set it on the service instance pointer - I think this would be a bit more readable. Also, it could be a static function, does not need to be a reconciler method - the function does not use anything from the receiver. 

TL;DR, something like

```
cfServiceInstance.Status.LastOperation = reconcileLastOperation(cfServiceInstance, credentialsSecret)
```